src/algorithms/ctmrg/sequential.jl

```diff
@@ -0,0 +1,109 @@
+"""
+    sequential_ctmrg_iter(state, envs::CTMRGEnv, alg::CTMRG) -> envs′, info
+
+Perform one sequential iteration of CTMRG, where one iteration consists of four expansion,
+renormalization and rotation steps that are performed sequentially.
+"""
+function sequential_ctmrg_iter(state, envs::CTMRGEnv, alg::CTMRG)
+    ϵ = zero(real(scalartype(state)))
+    for _ in 1:4 # rotate
+        for col in 1:size(state, 2) # left move column-wise
+            projectors, info = sequential_projectors(col, state, envs, alg.projector_alg)
+            envs = renormalize_sequentially(col, projectors, state, envs)
+            ϵ = max(ϵ, info.err)
+        end
+        state = rotate_north(state, EAST)
+        envs = rotate_north(envs, EAST)
+    end
+
+    return envs, (; err=ϵ)
+end
+
+"""
+    sequential_projectors(col::Int, state::InfinitePEPS, envs::CTMRGEnv, alg::ProjectorAlgs)
+    sequential_projectors(coordinate::NTuple{3,Int}, state::InfinitePEPS, envs::CTMRGEnv, alg::ProjectorAlgs)
+
+Compute CTMRG projectors in the `:sequential` scheme either for an entire column `col` or
+for a specific `coordinate` (where `dir=WEST` is already implied in the `:sequential` scheme).
+"""
+function sequential_projectors(
+    col::Int, state::InfinitePEPS, envs::CTMRGEnv, alg::ProjectorAlgs
+)
+    ϵ = zero(real(scalartype(envs)))
+
+    # SVD half-infinite environment column-wise
+    coordinates = eachcoordinate(envs)[:, col]
+    projectors = dtmap(coordinates) do (r, c)
+        proj, info = sequential_projectors((WEST, r, c), state, envs, alg)
+        ϵ = max(ϵ, info.err / norm(info.S))
```

Comment:
Just FYI, I noticed this before too, but this really shouldn't be working: we are updating epsilon in parallel, which might lead to race conditions etc. The solution is to either use an atomic variable, or just also return this value.

Reply:
Whoopsie, don't know why I missed that before. I quickly resolved that by also writing the errors into an pre-allocated Zygote Buffer.